codice/ddf
# DDF-4202 AbstractCswSource to transform Query Filter Properties based on typeName

#### What does this PR do?
AbstractCswSource to transform Query Filter Properties based on typeName
CSW Source need to be able to transform from internal query representation to another given query representation during federation. 
As part of a source configuration, typeName is provided. This should be used to look up the appropriate QueryFilterTransformer.

#### Who is reviewing it? 
@codice/ogc 
@bdeining 
@millerw8 
@mojogitoverhere 
@AzGoalie 
@troymohl 
@jlcsmith 

#### Ask 2 committers to review/merge the PR and tag them here.
@bdeining
@millerw8
@jlcsmith 

#### How should this be tested?
Ensure build and CI run successfully
Upload an image into DDF.
Create a CSW Specification federated source. Leave all default configuration
From UI, perform a query against this federated source and verify the result

#### Any background context you want to provide?
This is an architectural enabler for 
https://codice.atlassian.net/browse/DDF-4204
https://codice.atlassian.net/browse/DDF-4207

#### What are the relevant tickets?
https://codice.atlassian.net/browse/DDF-4202

#### Screenshots

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
